### PR TITLE
ENH: __array_ufunc__ handle np.minimum.reduce

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [x] whatsnew entry

np.mimum.reduce(frame) -> return Series instead of ndarray

np.maximum.reduce(series[dt64]) -> return Timestamp instead of np.datetime64
np.maximum.reduce(series[td64]) -> return Timedelta instead of np.timedelta64
np.maximum.reduce(DTA) -> return Timestamp instead of np.datetime64
np.maximum.reduce(TDA) -> return Timedelta instead of np.timedelta64
np.maximum.reduce(DTI) -> return Timestamp instead of np.datetime64
np.maximum.reduce(TDI) -> return Timedelta instead of np.timedelta64

np.maximum.reduce(integer_array) works instead of raising NotImplementedError
np.maximum.reduce(floating_array) works instead of raising NotImplementedError
np.maximum.reduce(Series[Int]) works instead of raising NotImplementedError
np.maximum.reduce(Series[Float]) works instead of raising NotImplementedError

Ditto for np.minimum.reduce

### CODE REVIEW

File: pandas/core/arrays/base.py [hunk 1]
```diff
@@ -1511,6 +1511,13 @@ def __array_ufunc__(self, ufunc: np.ufunc, method: str, *inputs, **kwargs):
                 self, ufunc, method, *inputs, **kwargs
             )
 
+        if method == "reduce":
+            result = arraylike.dispatch_reduction_ufunc(
+                self, ufunc, method, *inputs, **kwargs
+            )
+            if result is not NotImplemented:
+                return result
+
```
Comment (jreback) [inline pandas/core/arrays/base.py:1520]: can you not hook into the default_array_ufunc?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/arrays/base.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/arrays/base.py:1520]: thats the idea eventually

File: pandas/core/arraylike.py [hunk 2]
```diff
@@ -473,3 +488,39 @@ def default_array_ufunc(self, ufunc: np.ufunc, method: str, *inputs, **kwargs):
...
+
+    # NB: we are assuming that min/max represent minimum/maximum methods,
+    #  which would not be accurate for e.g. Timestamp.min
+    if not hasattr(self, method_name):
+        return NotImplemented
+
+    if self.ndim > 1:
+        if isinstance(self, ABCNDFrame):
+            # TODO: test cases where this doesn't hold, i.e. 2D DTA/TDA
+            kwargs["numeric_only"] = False
```
Comment (jreback) [inline pandas/core/arraylike.py:515]: do we normally modify the inputs?

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
needs a whatsnew note

File: pandas/core/arraylike.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/arraylike.py:515]: no, but that's bc the ufunc takes different kwargs than the DataFrame method

**Comment by jbrockmendel** [comment]
> needs a whatsnew note

has one

**Review by jreback — APPROVED** [review]

**Pull request merged**